contracts/note/src/state.rs

```diff
@@ -8,4 +9,23 @@ pub const CONNECTION_REMOTE_PORT: Item<(String, String)> = Item::new("a");
 pub const CHANNEL: Item<String> = Item::new("b");
 
 /// Max gas usable in a single block.
-pub(crate) const BLOCK_MAX_GAS: Item<u64> = Item::new("bmg");
+pub const BLOCK_MAX_GAS: Item<u64> = Item::new("bmg");
+
+/// (channel_id) -> sequence number. `u64` is the type used in the
+/// Cosmos SDK for sequence numbers:
+///
+/// <https://github.com/cosmos/ibc-go/blob/a25f0d421c32b3a2b7e8168c9f030849797ff2e8/modules/core/02-client/keeper/keeper.go#L116-L125>
+const SEQUENCE_NUMBER: Map<String, u64> = Map::new("sn");
+
+/// Increments and returns the next sequence number.
+pub(crate) fn increment_sequence_number(
+    storage: &mut dyn Storage,
+    channel_id: String,
+) -> StdResult<u64> {
+    let seq = SEQUENCE_NUMBER
+        .may_load(storage, channel_id.clone())?
+        .unwrap_or_default()
+        + 1;
```

Comment:
not sure if it's worth to do `checked_add` here and map to some specific error?

Reply:
oh wow, this is a great place for an error actually!